src/Modules/MatchRankingModule/Services/MatchRankingService.cs

```diff
@@ -0,0 +1,105 @@
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class MatchRankingService : IMatchRankingService
+{
+    private const int ShowRows = 4;
+
+    private readonly IManialinkManager _manialinkManager;
+    private readonly IPlayerManagerService _playerManager;
+    private readonly ILogger<MatchRankingService> _logger;
+    private readonly IEvoScBaseConfig _config;
+    private MatchRankingStore _matchRankingStore;
+
+    public MatchRankingService(IManialinkManager manialinkManager, IPlayerManagerService playerManager,
+        ILogger<MatchRankingService> logger, IEvoScBaseConfig config)
+    {
+        _manialinkManager = manialinkManager;
+        _playerManager = playerManager;
+        _logger = logger;
+        _config = config;
+        _matchRankingStore = new MatchRankingStore();
+    }
+
+    public Task UpdateAndShowScores(ScoresEventArgs scores)
+    {
+        _matchRankingStore.ConsumeScores(scores);
+        return SendManialink();
+    }
+
+    public async Task SendManialink()
+    {
+        _logger.LogInformation("Sending manialink.");
+
+        await _manialinkManager.SendPersistentManialinkAsync("MatchRankingModule.MatchRanking", GetWidgetData());
+    }
+
+    private dynamic GetWidgetData()
+    {
+        //TODO: improve so only relevant scores are mapped to save resources
+        var mappedScoresPrevious = MapScoresForWidget(_matchRankingStore.GetPreviousMatchScores()).ToList().Take(ShowRows).ToList();
+        var mappedScoresLatest = MapScoresForWidget(_matchRankingStore.GetLatestMatchScores()).ToList().Take(ShowRows).ToList();
+
+        var mappedScoresExisting = mappedScoresLatest
+            .Where(ranking => mappedScoresPrevious.Contains(ranking, new RankingComparer())).ToList();
+        var mappedScoresNew = mappedScoresLatest.Except(mappedScoresExisting).ToList();
+        
+        return new
+        {
+            NewScores = mappedScoresNew,
+            ExistingScores = mappedScoresExisting,
+            PreviousScores = mappedScoresPrevious,
+            headerColor = _config.Theme.UI.HeaderBackgroundColor,
+            primaryColor = _config.Theme.UI.PrimaryColor,
+            logoUrl = _config.Theme.UI.LogoWhiteUrl,
+            playerRowBackgroundColor = _config.Theme.UI.PlayerRowBackgroundColor
+        };
+    }
+
+    private IEnumerable<LiveRankingWidgetPosition> MapScoresForWidget(ScoresEventArgs? scores)
+    {
+        if (scores == null)
+        {
+            return new List<LiveRankingWidgetPosition>();
+        }
+
+        return scores.Players.ToList()
+            // .GetRange(0, ShowRows)
+            .Select(score => new LiveRankingWidgetPosition(
+                    score.Rank,
+                    _playerManager.GetPlayerAsync(score.AccountId).Result,
+                    (score.MatchPoints + score.RoundPoints).ToString(),
+                    -1,
+                    false
+                )
+            )
+            .ToList();
```

Comment:
converting to a list again seems redundant unless your goal here is to enumerate the list before returning